adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/CoapAdapterProperties.java

```diff
@@ -201,4 +235,122 @@ public final void setCoapThreads(final int threads) {
+     * <p>
+     * Timeout to wait before retransmit a flight, if no response is received.
+     * <p>
+     * The default value of this property is {@link #DEFAULT_DTLS_RETRANSMISSION_TIMEOUT},
+     * {@value #DEFAULT_DTLS_RETRANSMISSION_TIMEOUT} milliseconds.
+     * 
+     * @param dtlsRetransmissionTimeout timeout in milliseconds to retransmit a flight.
+     * @throws IllegalArgumentException if dtlsRetransmissionTimeout is &lt; 1.
+     */
+    public final void setDtlsRetransmissionTimeout(final int dtlsRetransmissionTimeout) {
+        if (dtlsRetransmissionTimeout < 1) {
+            throw new IllegalArgumentException("dtls retransmission timeout must be at least 1");
+        }
+        this.dtlsRetransmissionTimeout = dtlsRetransmissionTimeout;
+    }
+
+    /**
+     * Gets the exchange lifetime.
+     * <p>
+     * Time to keep coap request for deduplication.
+     * <p>
+     * The default value of this property is {@link #DEFAULT_EXCHANGE_LIFETIME}, {@value #DEFAULT_EXCHANGE_LIFETIME}
+     * milliseconds.
+     * 
+     * @return The exchange lifetime in milliseconds.
+     */
+    public final int getExchangeLifetime() {
+        return exchangeLifetime;
+    }
+
+    /**
+     * Gets the exchange lifetime.
+     * <p>
+     * Time to keep coap request for deduplication.
+     * <p>
+     * The default value of this property is {@link #DEFAULT_EXCHANGE_LIFETIME}, {@value #DEFAULT_EXCHANGE_LIFETIME}
+     * milliseconds.
+     * 
+     * @param exchangeLifetime the exchange lifetime in milliseconds to keep the request for deduplication.
+     * @throws IllegalArgumentException if exchangeLifetime is &lt; 1.
+     */
+    public final void setExchangeLifetime(final int exchangeLifetime) {
+        if (exchangeLifetime < 1) {
+            throw new IllegalArgumentException("exchange lifetime must be at least -1");
+        }
+        this.exchangeLifetime = exchangeLifetime;
+    }
+
+    /**
+     * Gets the message offloading mode.
+     * <p>
+     * When messages are kept for deduplication, parts of the message could be offloaded to reduce the heap consumption.
+     * <p>
+     * The default value of this property is {@link #DEFAULT_MESSAGE_OFFLOADING}, {@value #DEFAULT_MESSAGE_OFFLOADING}.
+     * 
+     * @return {@code true} enable message offloading, {@code false} disable message offloading.
+     */
+    public final boolean getMessageOffloading() {
+        return messageOffloading;
+    }
+
+    /**
+     * Sets the message offloading mode.
+     * <p>
+     * When messages are kept for deduplication, parts of the message could be offloaded to reduce the heap consumption.
+     * <p>
+     * The default value of this property is {@link #DEFAULT_MESSAGE_OFFLOADING}, {@value #DEFAULT_MESSAGE_OFFLOADING}.
+     * 
+     * @param messageOffloading {@code true} enable message offloading, {@code false} disable message offloading.
+     */
+    public final void setMessageOffloading(final boolean messageOffloading) {
```

Comment:
please rename to *setMessageOffloadingEnabled*

Reply:
done